src/components/Hero.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  Box,
+  Button,
+  type ButtonProps,
+  Stack,
+  Typography,
+  Container,
+  Grid,
+} from '@mui/material';
+import { type CommonProps, type Generic } from 'types/components';
+import { isJSX } from '../utils';
+
+export interface HeroProps extends CommonProps, HeroTextProps {
+  image?: string;
+  altText?: string;
+  inverse?: boolean;
+  background?: string;
+  size?: 'small' | 'big';
+  useHoverlay?: boolean;
+}
+
+interface CtaButton extends Partial<ButtonProps> {
+  text: string;
+}
+
+interface HeroTextProps extends CommonProps {
+  title: string;
+  subtitle?: string | Generic;
+  ctaButtons?: Array<CtaButton | Generic>;
+}
+
+const HeroTextContent = ({
+  title,
+  subtitle,
+  ctaButtons,
+  theme,
+}: HeroTextProps) => {
+  const textColor = theme === 'dark' ? 'primary.contrastText' : 'text.primary';
+  return (
+    <Stack
+      justifyContent={{ md: 'center' }}
+      sx={{ minHeight: 'inherit' }}
+      mt={{ xs: 9, lg: 0 }}
+    >
+      <Typography variant="h1" color={textColor} mb={2}>
+        {title}
+      </Typography>
+      <>
+        {subtitle && (
+          <Typography variant="body1" color={textColor}>
+            {subtitle}
+          </Typography>
+        )}
+      </>
+      {ctaButtons?.length ? (
+        <Stack
+          direction={{ xs: 'column', md: 'row' }}
+          spacing={2}
+          mt={4}
+          mb={{ xs: 4 }}
+        >
+          {ctaButtons.map((button, i) => {
+            if (isJSX(button)) return button;
+            return (
+              <Button key={`${button.text}-${i}`} {...button}>
+                {button.text}
+              </Button>
+            );
+          })}
+        </Stack>
+      ) : null}
+    </Stack>
+  );
+};
+
+const Hero = (props: HeroProps) => {
+  const {
+    size,
+    inverse = false,
+    background,
+    theme = 'dark',
+    useHoverlay = true,
+  } = props;
+  const minHeight = size === 'big' ? '720px' : '480px';
```

Comment:
Proposal: use MUI breakpoint instead explicit pixel sizes

Reply:
I'm not sure it's a good idea. First I see that default MUI breakpoints doesn't have 720 and 480 values. Breakpoints refers to  viewport's width, here I am setting a minimum height.